File: bindings/php/src/ProvenConfigmgmt.php
<?php
// SPDX-License-Identifier: PMPL-1.0-or-later
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// Config Mgmt protocol types for proven-servers.

declare(strict_types=1);

namespace ProvenServers;

/** ResourceType matching the Idris2 ABI tags. */
enum ResourceType: int
{
    case File = 0;
    case Package = 1;
    case Service = 2;
    case User = 3;
    case Group = 4;
    case Cron = 5;
    case Mount = 6;
    case Firewall = 7;
    case Registry = 8;
}

/** ResourceState matching the Idris2 ABI tags. */
enum ResourceState: int
{
    case Present = 0;
    case Absent = 1;
    case Running = 2;
    case Stopped = 3;
    case Enabled = 4;
    case Disabled = 5;
}

/** ChangeAction matching the Idris2 ABI tags. */
enum ChangeAction: int
{
    case Create = 0;
    case Modify = 1;
    case Delete = 2;
    case Restart = 3;
    case Reload = 4;
    case Skip = 5;
}

/** DriftStatus matching the Idris2 ABI tags. */
enum DriftStatus: int
{
    case InSync = 0;
    case Drifted = 1;
    case DUnknown = 2;
    case Unmanaged = 3;
}

/** ApplyMode matching the Idris2 ABI tags. */
enum ApplyMode: int
{
    case Enforce = 0;
    case DryRun = 1;
    case Audit = 2;
}

File: bindings/csharp/src/ProvenConfigmgmt.cs
// SPDX-License-Identifier: PMPL-1.0-or-later
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// Config Mgmt protocol types for proven-servers.

namespace Proven;

/// <summary>ResourceType matching the Idris2 ABI tags (0-8).</summary>
public enum ResourceType : byte
{
    File = 0,
    Package = 1,
    Service = 2,
    User = 3,
    Group = 4,
    Cron = 5,
    Mount = 6,
    Firewall = 7,
    Registry = 8
}

/// <summary>ResourceState matching the Idris2 ABI tags (0-5).</summary>
public enum ResourceState : byte
{
    Present = 0,
    Absent = 1,
    Running = 2,
    Stopped = 3,
    Enabled = 4,
    Disabled = 5
}

/// <summary>ChangeAction matching the Idris2 ABI tags (0-5).</summary>
public enum ChangeAction : byte
{
    Create = 0,
    Modify = 1,
    Delete = 2,
    Restart = 3,
    Reload = 4,
    Skip = 5
}

/// <summary>DriftStatus matching the Idris2 ABI tags (0-3).</summary>
public enum DriftStatus : byte
{
    InSync = 0,
    Drifted = 1,
    DUnknown = 2,
    Unmanaged = 3
}

/// <summary>ApplyMode matching the Idris2 ABI tags (0-2).</summary>
public enum ApplyMode : byte
{
    Enforce = 0,
    DryRun = 1,
    Audit = 2
}
